### PR TITLE
Remove manual classes for variations

### PR DESCRIPTION
**Pull request by boxadesign**
### What is the context of this PR?
The following components required the manual adding of classes to the `classes` param to use available variants. This is error prone and would produce unnecessary breaking changes in future:

- buttons
- lists
- input widths

Have introduced a new param to each component to set the variants. 

For buttons and lists the new param is `variants`. This accepts an array or single string value e.g. `'variants': ['small', 'secondary']` or `'variants': 'secondary'`.

For input widths the new parameter is `width`. e.g. `'width': '7@m'`. Previously the user had to use one or two classes (if number type or not) and this logic has now been automated in the macro (based on whether the type is number or tel). 

### What is the breaking change?

Developers will need to remove the manual classes being added to their component config and add the new appropriate param and value.

### How to review
Check everything is ok.

### CODE REVIEW

**Comment by jrbarnes9** [comment]
Needs a Percy check

File: src/patterns/correct-errors/examples/errors-proto/errors/index.njk [hunk 1]
```diff
@@ -126,7 +126,7 @@ layout: none
             onsInput({
                 "type": "number",
                 "id": "number-input",
-                "classes": "ons-input--w-2",
+                "width": "2",
```
Comment (jrbarnes9) [inline src/patterns/correct-errors/examples/errors-proto/errors/index.njk:129]:
```suggestion
                "width": "3",
```

**Comment by rmccar** [comment]
Looks good once Jons stuff is sorted

**Comment by jrbarnes9** [comment]
Needs to be updated to use width param? screenshot


<details><summary>Browser metadata</summary>





```
Path:      /patterns/question/
Browser:   Chrome 93.0.4577.63 on Mac OS 10.14.6
Viewport:  1542 x 978 @1x
Language:  en
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-1684--ons-design-system.netlify.app%2Fpatterns%2Fquestion%2F&os=OS%20X&os_version=Mojave&browser=Chrome&browser_version=93.0&resolution=1542x978&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-1684--ons-design-system.netlify.app/patterns/question/) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=614880896533e30007302ad9&commentID=61489521af858f14405dff81&resolution=resolved)

**Comment by jrbarnes9** [comment]
Worth updating this dates documentation to mention width param instead of class? screenshot


<details><summary>Browser metadata</summary>





```
Path:      /patterns/dates/
Browser:   Chrome 93.0.4577.63 on Mac OS 10.14.6
Viewport:  1542 x 978 @1x
Language:  en
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-1684--ons-design-system.netlify.app%2Fpatterns%2Fdates%2F&os=OS%20X&os_version=Mojave&browser=Chrome&browser_version=93.0&resolution=1542x978&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-1684--ons-design-system.netlify.app/patterns/dates/) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=614880896533e30007302ad9&commentID=6148957aaf858f1a8b5dfd14&resolution=resolved)

**Review by LauraC119 — CHANGES_REQUESTED** [review]
Those changes to the Width constrained section have made it even longer. 
Can you you cut the following: 
"Inputs can have their widths constrained (but not limited) to the number of expected characters using the width parameter."
and
"If you set type: text and width: 10 this will constrain the input to the width of 10 letters."

**Review by LauraC119 — DISMISSED** [review]

**Review by LauraC119 — APPROVED** [review]
OK, I see you made Jon's change. All good from me

**Review by jrbarnes9 — APPROVED** [review]

**Pull request merged**